### PR TITLE
Generate temp targets

### PR DESCRIPTION
**Pull request by Shalsh23**
Output results to temporary files before moving them to target files. 
Will prevent corrupted ghost files from messing up with make.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -95,7 +98,8 @@ $(BUILD_DIR)/ontology-metadata.ttl: $(BIO-ONTOLOGIES) $(SPARQL)/ontology-version
 	$(ROBOT) query \
 	-i $< \
 	--use-graphs true \
-	--query $(SPARQL)/ontology-versions.sparql $@
+	--query $(SPARQL)/ontology-versions.sparql $@.tmp \
```
Comment (balhoff) [inline Makefile]: Are you certain this will output turtle?

File: Makefile [hunk 2]
```diff
@@ -81,7 +83,8 @@ $(BUILD_DIR)/phenoscape-kb.ttl: $(BUILD_DIR)/ontology-metadata.ttl \
 
 # Compute Tbox hierarchy
 $(BUILD_DIR)/phenoscape-kb-tbox-hierarchy.ttl: $(BUILD_DIR)/phenoscape-kb-tbox-classified.ttl $(SPARQL)/subclassHierarchy.sparql
-	$(ARQ) --data=$< --query=$(SPARQL)/subclassHierarchy.sparql > $@
+	$(ARQ) --data=$< --query=$(SPARQL)/subclassHierarchy.sparql > $@.tmp \
```
Comment (balhoff) [inline Makefile]: We should specify the output format to ensure we get turtle.

File: Makefile [hunk 3]
```diff
@@ -289,14 +316,16 @@ $(BUILD_DIR)/anatomical-entities.txt: $(BUILD_DIR)/bio-ontologies-classified.ofn
 	$(ROBOT) query \
     	-i $< \
     	--use-graphs true \
-    	--query $(SPARQL)/anatomicalEntities.sparql $@
+    	--query $(SPARQL)/anatomicalEntities.sparql $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 4]
```diff
@@ -289,14 +316,16 @@ $(BUILD_DIR)/anatomical-entities.txt: $(BUILD_DIR)/bio-ontologies-classified.ofn
...
+    	&& mv $@.tmp $@
 
 # Generate qualities.txt
 $(BUILD_DIR)/qualities.txt: $(BUILD_DIR)/bio-ontologies-classified.ofn $(SPARQL)/qualities.sparql
 	$(ROBOT) query \
     	-i $< \
     	--use-graphs true \
-    	--query $(SPARQL)/qualities.sparql $@
+    	--query $(SPARQL)/qualities.sparql $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 5]
```diff
@@ -375,21 +416,25 @@ $(BUILD_DIR)/hpoa.ttl:
 $(BUILD_DIR)/gene-profiles.ttl: $(BUILD_DIR)/monarch-data-merged.ttl $(SPARQL)/geneProfiles.sparql
 	$(ROBOT) query \
     	-i $< \
-    	--query $(SPARQL)/geneProfiles.sparql $@
+    	--query $(SPARQL)/geneProfiles.sparql $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 6]
```diff
@@ -375,21 +416,25 @@ $(BUILD_DIR)/hpoa.ttl:
...
+    	&& mv $@.tmp $@
 
 # Generate absences.ttl
 $(BUILD_DIR)/absences.ttl: $(BUILD_DIR)/phenex-data+tbox.ttl $(SPARQL)/absences.sparql
-	$(ARQ) --data=$< --query=$(SPARQL)/absences.sparql > $@
+	$(ARQ) --data=$< --query=$(SPARQL)/absences.sparql > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 7]
```diff
@@ -375,21 +416,25 @@ $(BUILD_DIR)/hpoa.ttl:
...
+	&& mv $@.tmp $@
 
 # Generate presences.ttl
 $(BUILD_DIR)/presences.ttl: $(BUILD_DIR)/phenex-data+tbox.ttl $(SPARQL)/presences.sparql
-	$(ARQ) --data=$< --query=$(SPARQL)/presences.sparql > $@
+	$(ARQ) --data=$< --query=$(SPARQL)/presences.sparql > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 8]
```diff
@@ -403,13 +448,15 @@ $(BUILD_DIR)/evolutionary-profiles.ttl: $(BUILD_DIR)/phenex-data+tbox.ttl
 $(BUILD_DIR)/subclass-closure.ttl: $(BUILD_DIR)/phenoscape-kb-tbox-classified.ttl $(SPARQL)/subclass-closure-construct.sparql
 	$(ARQ) \
 	--data=$< \
-	--query=$(SPARQL)/subclass-closure-construct.sparql > $@
+	--query=$(SPARQL)/subclass-closure-construct.sparql > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 9]
```diff
@@ -403,13 +448,15 @@ $(BUILD_DIR)/evolutionary-profiles.ttl: $(BUILD_DIR)/phenex-data+tbox.ttl
...
+	&& mv $@.tmp $@
 
 # Compute instance closures
 $(BUILD_DIR)/instance-closure.ttl: $(BUILD_DIR)/phenex-data+tbox.ttl $(SPARQL)/profile-instance-closure-construct.sparql
 	$(ARQ) \
 	--data=$< \
-	--query=$(SPARQL)/profile-instance-closure-construct.sparql > $@
+	--query=$(SPARQL)/profile-instance-closure-construct.sparql > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 10]
```diff
@@ -464,47 +513,56 @@ $(BUILD_DIR)/gene-pairwise-sim.ttl: $(BUILD_DIR)/profiles.ttl $(BUILD_DIR)/pheno
...
+	python $(RESOURCES)/regression.py `grep -v 'VTO_' $(BUILD_DIR)/profile-sizes.txt | wc -l` $< $@.tmp \
+	&& mv $@.tmp $@
 
 $(BUILD_DIR)/taxa-scores.tsv: $(SPARQL)/get-scores.rq $(BUILD_DIR)/corpus-ics-taxa.ttl $(BUILD_DIR)/taxa-pairwise-sim.ttl
 	$(ARQ) \
 	--data=$(BUILD_DIR)/corpus-ics-taxa.ttl \
 	--data=$(BUILD_DIR)/taxa-pairwise-sim.ttl \
 	--results=TSV \
-	--query=$< > $@
+	--query=$< > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

File: Makefile [hunk 11]
```diff
@@ -464,47 +513,56 @@ $(BUILD_DIR)/gene-pairwise-sim.ttl: $(BUILD_DIR)/profiles.ttl $(BUILD_DIR)/pheno
...
+	python $(RESOURCES)/regression.py `grep -v 'VTO_' $(BUILD_DIR)/profile-sizes.txt | wc -l` $< $@.tmp \
+	&& mv $@.tmp $@
 
 $(BUILD_DIR)/gene-scores.tsv: $(SPARQL)/get-scores.rq $(BUILD_DIR)/corpus-ics-genes.ttl $(BUILD_DIR)/gene-pairwise-sim.ttl
 	$(ARQ) \
 	--data=$(BUILD_DIR)/corpus-ics-genes.ttl \
 	--data=$(BUILD_DIR)/gene-pairwise-sim.ttl \
 	--results=TSV \
-	--query=$< > $@
+	--query=$< > $@.tmp \
```
Comment (balhoff) [inline Makefile]: Specify out format if possible.

**Comment by Shalsh23** [comment]
@balhoff Ready to merge.

**Pull request merged**